bug: Flaky Snowplow tracker test

### Meltano Version

NA

### Python Version

3.13

### Bug scope

CLI (options, error messages, logging, etc.)

### Operating System

NA

### Description

https://github.com/meltano/meltano/blob/47d390e7625cdf7b13c0c02e31df1b45f325bd85/tests/meltano/core/tracking/test_tracker.py#L459

```python
self = <test_tracker.TestTracker object at 0x7f2df2e531f0>, project = <meltano.core.project.Project object at 0x7f2dedee6b30>
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f2dee0f4f40>, log = <pytest_structlog.StructuredLogCapture object at 0x7f2dee0f4fd0>

    @pytest.mark.order(1)
    def test_get_snowplow_tracker_invalid_endpoint(
        self,
        project: Project,
        monkeypatch: pytest.MonkeyPatch,
        log: pytest_structlog.StructuredLogCapture,
    ) -> None:
        endpoints = """
            [
                "notvalid:8080",
                "https://valid.endpoint:8080/",
                "file://bad.scheme",
                "https://other.endpoint/path/to/collector"
            ]
        """
        monkeypatch.setenv("MELTANO_SNOWPLOW_COLLECTOR_ENDPOINTS", endpoints)
    
        tracker = Tracker(project)
    
        try:
>           assert log.events[0]["level"] == "warning"
E           IndexError: list index out of range

/home/runner/work/meltano/meltano/tests/meltano/core/tracking/test_tracker.py:459: IndexError
```

### References

- https://github.com/meltano/meltano/actions/runs/12837934995/job/35802576758?pr=8980

### Code

```python

```

### Link to Slack/Linen

_No response_

--- a/src/meltano/cli/__init__.py
+++ b/src/meltano/cli/__init__.py
@@ -2,6 +2,7 @@
 
 from __future__ import annotations
 
+import logging
 import os
 import sys
 import typing as t
@@ -119,6 +120,7 @@ def _run_cli() -> None:
 def main() -> None:
     """Entry point for the meltano CLI."""
     # Mark the current process as executed via the CLI
+    logging.captureWarnings(capture=True)
     os.environ["MELTANO_JOB_TRIGGER"] = os.getenv("MELTANO_JOB_TRIGGER", "cli")
     try:
         _run_cli()

--- a/src/meltano/core/tracking/tracker.py
+++ b/src/meltano/core/tracking/tracker.py
@@ -125,7 +125,11 @@ class Tracker:  # - too many (public) methods
         emitters: list[Emitter] = []
         for endpoint in endpoints:
             if not check_url(endpoint):
-                logger.warning("invalid_snowplow_endpoint", endpoint=endpoint)
+                warn(
+                    f"Invalid Snowplow endpoint: {endpoint}",
+                    UserWarning,
+                    stacklevel=2,
+                )
                 continue
             parsed_url = urlparse(endpoint)
             emitters.append(

--- a/tests/meltano/core/tracking/test_tracker.py
+++ b/tests/meltano/core/tracking/test_tracker.py
@@ -1,7 +1,6 @@
 from __future__ import annotations
 
 import json
-import logging
 import subprocess
 import typing as t
 import uuid
@@ -12,11 +11,8 @@ from time import sleep
 
 import mock
 import pytest
-import structlog
 from snowplow_tracker import Emitter, SelfDescribing
 
-from meltano.core.logging import setup_logging
-from meltano.core.project import Project
 from meltano.core.tracking.contexts.cli import CliEvent
 from meltano.core.tracking.contexts.environment import EnvironmentContext
 from meltano.core.tracking.contexts.exception import ExceptionContext
@@ -26,11 +22,9 @@ from meltano.core.utils import hash_sha256
 
 if t.TYPE_CHECKING:
     from collections.abc import Generator
-    from pathlib import Path
-
-    import pytest_structlog
 
     from fixtures.docker import SnowplowMicro
+    from meltano.core.project import Project
 
 
 def load_analytics_json(project: Project) -> dict[str, t.Any]:
@@ -443,67 +437,40 @@ class TestTracker:
     @pytest.mark.order(1)
     def test_get_snowplow_tracker_invalid_endpoint(
         self,
-        tmp_path: Path,
+        project: Project,
         monkeypatch: pytest.MonkeyPatch,
-        log: pytest_structlog.StructuredLogCapture,
     ) -> None:
-        monkeypatch.delenv("MELTANO_CLI_LOG_CONFIG", raising=False)
-        monkeypatch.delenv("MELTANO_CLI_LOG_LEVEL", raising=False)
-        monkeypatch.delenv("MELTANO_SNOWPLOW_COLLECTOR_ENDPOINTS", raising=False)
-        # Set up a new project just for this test
-        endpoints = [
-            "notvalid:8080",
-            "https://valid.endpoint:8080",
-            "file://bad.scheme",
-            "https://other.endpoint/path/to/collector",
-        ]
-        meltano_yml = {
-            "project_id": "test-project",
-            "disable_tracking": True,
-            "cli": {
-                "log_level": "warning",
-            },
-            "snowplow": {
-                "collector_endpoints": endpoints,
-            },
-        }
-        tmp_path.joinpath("meltano.yml").write_text(json.dumps(meltano_yml))
-        project = Project(tmp_path)
-        setup_logging(project=project)
+        endpoints = """
+            [
+                "notvalid:8080",
+                "https://valid.endpoint:8080",
+                "file://bad.scheme",
+                "https://other.endpoint/path/to/collector"
+            ]
+        """
+        monkeypatch.setenv("MELTANO_SNOWPLOW_COLLECTOR_ENDPOINTS", endpoints)
 
-        tracker = Tracker(project)
+        with pytest.warns(
+            (UserWarning, UserWarning),
+            match="Invalid Snowplow endpoint",
+        ) as warnings:
+            tracker = Tracker(project)
 
-        assert project.settings.get("cli.log_level") == "warning"
-        assert project.settings.get("snowplow.collector_endpoints") == endpoints
+        assert warnings[0].message.args[0] == "Invalid Snowplow endpoint: notvalid:8080"
+        assert (
+            warnings[1].message.args[0]
+            == "Invalid Snowplow endpoint: file://bad.scheme"
+        )
 
-        logger = structlog.stdlib.get_logger("meltano.core.tracking.tracker")
-        assert logger.isEnabledFor(logging.WARNING)
+        assert len(tracker.snowplow_tracker.emitters) == 2
 
-        try:
-            assert log.events[0]["level"] == "warning"
-            assert log.events[0]["event"] == "invalid_snowplow_endpoint"
-            assert log.events[0]["endpoint"] == "notvalid:8080"
+        emitter = tracker.snowplow_tracker.emitters[0]
+        assert isinstance(emitter, Emitter)
+        assert emitter.endpoint.startswith("https://valid.endpoint:8080/")
 
-            assert log.events[1]["level"] == "warning"
-            assert log.events[1]["event"] == "invalid_snowplow_endpoint"
-            assert log.events[1]["endpoint"] == "file://bad.scheme"
-
-            assert len(tracker.snowplow_tracker.emitters) == 2
-
-            emitter = tracker.snowplow_tracker.emitters[0]
-            assert isinstance(emitter, Emitter)
-            assert emitter.endpoint.startswith(
-                "https://valid.endpoint:8080/",
-            )
-
-            emitter = tracker.snowplow_tracker.emitters[1]
-            assert isinstance(emitter, Emitter)
-            assert emitter.endpoint.startswith(
-                "https://other.endpoint/path/to/collector/",
-            )
-        finally:
-            # Remove the seemingly valid emitters to prevent a logging error on exit.
-            tracker.snowplow_tracker.emitters = []
+        emitter = tracker.snowplow_tracker.emitters[1]
+        assert isinstance(emitter, Emitter)
+        assert emitter.endpoint.startswith("https://other.endpoint/path/to/collector/")
 
     def test_client_id_from_env_var(
         self,

--- a/tests/meltano/core/tracking/test_tracker.py
+++ b/tests/meltano/core/tracking/test_tracker.py
@@ -1,6 +1,7 @@
 from __future__ import annotations
 
 import json
+import logging
 import subprocess
 import typing as t
 import uuid
@@ -11,8 +12,11 @@ from time import sleep
 
 import mock
 import pytest
+import structlog
 from snowplow_tracker import Emitter, SelfDescribing
 
+from meltano.core.logging import setup_logging
+from meltano.core.project import Project
 from meltano.core.tracking.contexts.cli import CliEvent
 from meltano.core.tracking.contexts.environment import EnvironmentContext
 from meltano.core.tracking.contexts.exception import ExceptionContext
@@ -22,11 +26,11 @@ from meltano.core.utils import hash_sha256
 
 if t.TYPE_CHECKING:
     from collections.abc import Generator
+    from pathlib import Path
 
     import pytest_structlog
 
     from fixtures.docker import SnowplowMicro
-    from meltano.core.project import Project
 
 
 def load_analytics_json(project: Project) -> dict[str, t.Any]:
@@ -439,21 +443,41 @@ class TestTracker:
     @pytest.mark.order(1)
     def test_get_snowplow_tracker_invalid_endpoint(
         self,
-        project: Project,
+        tmp_path: Path,
         monkeypatch: pytest.MonkeyPatch,
         log: pytest_structlog.StructuredLogCapture,
     ) -> None:
-        endpoints = """
-            [
-                "notvalid:8080",
-                "https://valid.endpoint:8080",
-                "file://bad.scheme",
-                "https://other.endpoint/path/to/collector"
-            ]
-        """
-        monkeypatch.setenv("MELTANO_SNOWPLOW_COLLECTOR_ENDPOINTS", endpoints)
+        monkeypatch.delenv("MELTANO_CLI_LOG_CONFIG", raising=False)
+        monkeypatch.delenv("MELTANO_CLI_LOG_LEVEL", raising=False)
+        monkeypatch.delenv("MELTANO_SNOWPLOW_COLLECTOR_ENDPOINTS", raising=False)
+        # Set up a new project just for this test
+        endpoints = [
+            "notvalid:8080",
+            "https://valid.endpoint:8080",
+            "file://bad.scheme",
+            "https://other.endpoint/path/to/collector",
+        ]
+        meltano_yml = {
+            "project_id": "test-project",
+            "disable_tracking": True,
+            "cli": {
+                "log_level": "warning",
+            },
+            "snowplow": {
+                "collector_endpoints": endpoints,
+            },
+        }
+        tmp_path.joinpath("meltano.yml").write_text(json.dumps(meltano_yml))
+        project = Project(tmp_path)
+        setup_logging(project=project)
 
         tracker = Tracker(project)
+
+        assert project.settings.get("cli.log_level") == "warning"
+        assert project.settings.get("snowplow.collector_endpoints") == endpoints
+
+        logger = structlog.stdlib.get_logger("meltano.core.tracking.tracker")
+        assert logger.isEnabledFor(logging.WARNING)
 
         try:
             assert log.events[0]["level"] == "warning"

